### DISCUSSION TITLE
Allow multiple correspondents for a given document

### DISCUSSION THREAD

**Discussion by Kazer67** [paperless-ngx/paperless-ngx#437] [Feature Requests]
Since many persons / company may work on a given document, the ability to set more than one Correspondents like Tags would be a good feature.

**Comment by sukisoft** [comment]
I think, this could be done via a Tag which both persons are monitoring through a stored view.

**Reply by Kazer67** [reply]
Yes, obviously the tag system let you put what you want but since there's a "correspondents" feature, may as well use it for that purpose.

Since you may have a lot of correspondants and tag, separating those two seem useful (which is the case for my dad since he's a woodcrafter and have a lot of correspondent as customer).

**Reply by sukisoft** [reply]
I see your point. For my workflow this wouldnt be needed, but lets wait and see if others would want this, too.

**Reply by Kazer67** [reply]
That's also why I made another feature request (https://github.com/paperless-ngx/paperless-ngx/discussions/436) to improve the correspondents feature.

For some of my document concerning gouvernement stuff, having the ability to add a correspondent and in that correspondent "profil" thing like e-mail or number would help greatly when going back to it and not having to search who was the contact for this specific documents.

It's feedback from my dad's use case and my own but the correspondents feature could be really useful with some change as right now, it's close to just be a kind of "tag".

**Reply by sukisoft** [reply]
Yep, custom metadata to some objects lately is a feature which has been asked for quite a few times. See #371 and your #436. There are even more i think.

Should be something, most people want i guess. But has to be prepared and reviewed before someone should star implement it without an idea to what this feature should lead in the end.

**Comment by egore** [comment]
Multiple correspondents would be very useful when dealing with multiple legal entities. One example is contractual documents: They usually involve two parties, which would both be represented best as correspondents. If additionally also manage multiple legal entities within one installation, it quickly renders the correspondents feature hard to use: Which party would you use as correspondent, which party as tag?

**Comment by NetHans** [comment]
It might be useful to look at the correspondents in a more abstract way. As a rule, you communicate with companies, so many correspondents currently only have the general company name and the persons as a tag. Whereby the tag is also very often left out. It could be useful to design the correspondent as a tree. So that for example in the first level the company, in the second level the department and in the third level the person can be created and grouped. This tree should of course be variable in depth. My guess is that in most cases the tree will have a maximum of 2 to 3 levels. That would still be clear.

**Comment by PN3UMA** [comment]
My organization's use case is that we process memberships in batches, so multiple people are associated with one document. It would be great if that could be reflected in Paperless.

**Comment by Joerg-L** [comment]
We would also appreciate to have the option to add multiple correspondents to a document.

From documentation
> The correspondent of a document is the person, institution or company that a document either originates from, or is sent to.

We don't see the benefit of "or" on that feature.

Our use case:
We want to use paperless ngx for our whole family and it would be really good to be able to query for originator AND receiver and not only for one of it. And using "correspondents" is obviously the right approach for that.

**Comment by EdMueller** [comment]
> We want to use paperless ngx for our whole family and it would be really good to be able to query for originator AND receiver and not only for one of it. And using "correspondents" is obviously the right approach for that.

We too ;-).

In my usecase, i need a **sender** correspondent and a **recipient** correspondent. And the **recipient** correspondent should be one of a **hierachical tree**, e.g.
- Family
  - Me
  - Father
  - Mother
  - Wife
  - Child
  - Stepmum
  - ...

If implemented (please), i would use **hierachical tags** instead.

**Comment by Infouzi** [comment]
Would be also very interested in this too
- Having "sender" (which is the actual correspondent field) and "receiver"
- The receiver can be a "multiple" type of field, because the same document can be sent to multiple entities to deal with

**Comment by krehl** [comment]
Would love to see a solution based on custom fields. If possible a custom field of type correspondent would solve almost all use cases.

**Reply by dichkovsky** [reply]
I'll second that

**Reply by chris-cgn** [reply]
I'd prefer custom field with type correspondent (gives extra 'labels'), multiple entries in correspondent field would be helpful as well (and maybe better in other use-cases described above like multiple recipients).

My use-case: medical bill management.
Most doctors do not send out their own invoices, but rather hire companies to bill their patients. I usually handle all payment through these companies, but in case of errors I have to communicate with all three parties.

**Reply by andzno1** [reply]
I would really like to see this one happening. It might even be a feature request different from "multiple correspondents".

**Reply by ce603** [reply]
One way to make this work would be an "additional correspondents" field, which would pull from the correspondents table. In search results filtering by correspondent, you would see the results for documents where the entity was a primary or additional correspondent. 

More generally, if you could create a new custom field and set the data type to an existing field (e.g. Correspondent, Document Type), I could see this being helpful. 

This works around the limitation where you can set the file path based on {{ correspondent }}, which would get screwed up if you have Correspondent A, Correspondent B; it would probably be best to have the folder named after the primary correspondent.

**Comment by BubbleRep** [comment]
a custom field with type correspondent would be awesome.

**Comment by gerhardt** [comment]
I believe this feature would just imply to set the 'correspondent' field to a 'multiple' field, such as the 'references' field?! I.e. I strongly support this feature request.

**Comment by TheInfamousAlk** [comment]
I would also use this. For example, on my joint tax return, I had to choose just one correspondent, instead of me, my wife, and my tax person to whom I sent the document all at once. For now I am going to use tags, but as others said that's not ideal.

**Comment by madduck** [comment]
Yes, please.
(also helping this issue not get archived)

**Comment by ShapeShifter499** [comment]
Adding my two cents here to keep this open. I have documents that I think would qualify for multiple correspondents. I would like to see a option to select multiple correspondents.

**Comment by HiranChaudhuri** [comment]
Same here.

Currently I am using Correspondent more like "Author", which is only one entity for each document. But there are many entities I share a document with - and those shall be reflected in my correspondence. How often would I request a statement from one side to present it to another? Yes, I am using some documents in multiple conversations with different correspondents.

Please allow multiple correspondents per document.

**Comment by ZenDevMaster** [comment]
Jumping into the fray - found this discussion when searching "paperless ngx multiple correspondents" and bummed to see it as a 3 year old feature request. 

Here's a major use case:

1. Any 3 or more party document

When I pay someone with my bank app, the bank transfer generates a payment slip document. The first correspondent is the bank  itself, but a second - equally crucial (for searching purposes) - detail is who I sent the payment to - which should also be a correspondent. This would let us accurately filter or search for all docs from that bank OR all docs related to that other person and both would return the same document. 

I can think of a lot of 3 or more party documents... mortgage servicing company & originating bank, travel reservations that involve an agent, most legal paperwork (i.e. every party + attorney), a shipping label, etc.

The UX for this could switch from the Correspondent drop-down to the "Tags" style conbo-box selector and still pull from the Correspondent model. It looks like the correspondent field uses a ForeignKey while tags is a ManyToManyField on the Document model. Advanced search already implements searching Tags, so perhaps some of the groundwork is already laid if the project managers approve such a switch. Not sure about how the classifier or other components might need to be reworked.

**Comment by gerhardt** [comment]
Although, I was very positive about this feature before, I am unsure how this can be exactly implemented: If you have the files in the filesystem, some people would like to have then ordered by a (single) correspondent - i.e. all the stuff of my customs are in one place.
How to deal with this when 'multiple correspondents' are allowed?

**Reply by TheInfamousAlk** [reply]
One solution to that would be a filter that lets you select one or more correspondents and it will filter items to only those with one or more of the selected correspondents. And another separate filter (or an AND/OR toggle) to sort by only items with ALL selected correspondents. Something like this: 
![image](https://github.com/user-attachments/assets/3cb2926b-9be6-4912-8074-bcb65b5e6caf)

**Reply by BubbleRep** [reply]
I guess he meant how the file would be put in the filesystem ../correspondent1/.. or ../correspondent2/.. or ../correspondent1_correspondent2/.., I think the most elegant way would be to have them at ../correspondentX/.. with hard links to the first correspondent. But that would require that the underlaying filesystem supports hard links, otherwise it should fall back to copy (maybe optional) and have it in every correspondent folder.

**Reply by ce603** [reply]
I suggest a document have primary and additional correspondents. The document would be stored under the primary correspondent in the file system, but would match search results any time one of the additional correspondents was queried.

**Comment by glockmane** [comment]
Is this still being reviewed? I think there would be zero draw back to allow multiple correspondents? Nothing would break, anyone could use it like before but one could add more correspondents if he wants..

Let me give you an example:

I work for Company "A" and have a Child "B". Now my Child is sick and I need to go to the Doctor "C", who then gives me a attestation destined to my Insurance Company "D"..

I could set the correspondents to A,B,C,D in this case.. I could imagine using Tags "work","insurance" or maybe some others additionally..

**Reply by ce603** [reply]
One thing that may break is the storage paths... you can store a document based on {{ correspondent }}. Not sure how that would work when you have multiple correspondents. I suggested having primary and additional correspondents, and basing the storage path on the primary.

**Reply by HiranChaudhuri** [reply]
If we had a field 'author', then you would switch the storage path from correspondent to author. That one never changes, even if the document is used in another conversation with another correspondent.

Just imagine you get some statement from a bank and need to show it to the tax office.

**Reply by madduck** [reply]
> If we had a field 'author', then you would switch the storage path from correspondent to author. That one never changes, even if the document is used in another conversation with another correspondent.
> 
> Just imagine you get some statement from a bank and need to show it to the tax office.

While I'd say that when I show a document to the tax office, I don't actually store that document in the DMS again, I know what you mean.

This is an excellent proposal: s/correspondent/author/ and add an additional field "correpondents" such that when Alice writes a document and sends it to Bob with myself on CC, I can filter on Alice or Bob to see the document after I imported it.
